crates/re_types_builder/src/codegen/rust/deserializer.rs

```diff
@@ -654,9 +654,7 @@ fn quote_arrow_field_deserializer(
                                 // .collect::<DeserializationResult<Vec<_>>>()?;
 
                                 // NOTE: Unwrapping cannot fail: the length must be correct.
-                                let arr = array_init::from_iter(data).unwrap();
-
-                                Ok(arr)
+                                array_init::from_iter(data).ok_or(DeserializationError::array_init_underrun(#length))
```

Comment:
given that it was documented as such (and still is - comment outdated as it stands ;)), I believe we should keep the unwrap here and emit a localized `#[allow(...)]` with a comment on why.